app/logic/Mail/EWS/EWSFolder.ts

```diff
@@ -100,13 +101,13 @@ export class EWSFolder extends Folder {
         this.forEachSyncChange(result.Changes.Update, this.processSyncUpdate, false),
         this.forEachSyncChange(result.Changes.Create, this.processSyncUpdate, false),
       ])).flat();
-      let newMsgs = await this.getNewMessageHeaders(newMessageIDs);
```

Comment:
With the old code, I only used the new messages once, which was to add them to the message list inside the loop. But now you want all of the new messages, I need to collect them up, so that I can return them outside of the loop.

The alternative would be to collect up the new message IDs in the loop and then add a second loop in `getNewMessageHeaders` so that i only download 50 messages per API call.